### PR TITLE
Show protocol mismatch warning when request is https

### PR DESCRIPTION
**Pull request by oliverguenther**

### CODE REVIEW

File: app/views/warning_bar/_protocol_mismatch.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div id="pending-migrations-warning" class="warning-bar--item unsupported-browser-warning--pane">
+  <span class="icon3 icon-warning"></span>
+  <p>
+    <strong>Protocol setting mismatch</strong>
+    <br/>
+    Your application is running with Protocol set to <code><%= Setting.protocol %></code>, but the request
+    is a <code><%= request.ssl? ? 'https' : 'http' %></code>  request.
```
Comment (ulferts) [inline app/views/warning_bar/_protocol_mismatch.erb]: typo: a -> an (I think). Although 'h' is not a vowel, it is pronounced as such, almost like an 'a', in the context of 'http/s'.

The whole partial is not i18ned

**Review by ulferts — CHANGES_REQUESTED** [review]
I like the proposed change. Should help users running into less bugs. We might want to extend this to also include the hostname as it also causes a whole series of bugs, but that can also be done in a different PR.

Please see my i18n remark for the necessary changes.

File: app/views/warning_bar/_protocol_mismatch.erb (hunk 1 above)
Comment (oliverguenther) [reply app/views/warning_bar/_protocol_mismatch.erb]: I actually wanted to skip I18n here because it's admin stuff anyway, but fine for me to translate it

**Comment by oliverguenther** [comment]
I addressed your remarks. @machisuji @crohr Do you foresee any situations where the hostname of a browser request would not match the host_name setting ? The internal host name of a server obviously might differ, but the externally reachable address should always be the same right?

**Review by ulferts — APPROVED** [review]

**Pull request merged**